File: Scripting-Clase 1/Entity.cpp
#include "pch.h"
#include "Entity.h"

Entity::Entity()
{
	moveSpeed = 50;
}

Entity::~Entity()
{
}

void Entity::InitSprite(sf::Texture& _texture)
{
	_sprite.setTexture(_texture);
	
}

void Entity::Render(sf::RenderTarget* _target)
{
	_target->draw(_sprite);
}

void Entity::SetPosition(const float _x, const float _y)
{
	_sprite.setPosition(sf::Vector2f(_x, _y));
}

void Entity::Move(const float x, const float y, const float& dt)
{
	float xvalue = 0;
	xvalue += x * moveSpeed * dt;
	posX += x;
	float yvalue = 0;
	yvalue += y * moveSpeed * dt;
	posY += y;
	_sprite.move(sf::Vector2f(xvalue, yvalue));
}

void Entity::Update(const float& dt)
{

}

File: Scripting-Clase 1/Entity.lua
Entity = {}
Entity.__index = Entity

function Entity:New(num)
	local this =
	{
		myNum = num
	}
	setmetatable(this,self)
	return this
end

function Entity:print()
	print("Entity")
end


function Entity:InitSprite(_texture)
end

function Entity:SetPosition(_x, _y)
	
end
